remote/remote.go

```diff
@@ -347,12 +347,36 @@ func (i *Image) Name() string {
 }
 
 func (i *Image) Found() bool {
-	ref, auth, err := referenceForRepoName(i.keychain, i.repoName)
+	_, err := i.headManifest()
+	return err == nil
+}
+
+func (i *Image) CheckReadWriteAccess() bool {
+	ref, _, err := referenceForRepoName(i.keychain, i.repoName)
 	if err != nil {
 		return false
 	}
-	_, err = remote.Head(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport))
-	return err == nil
+	return remote.CheckPushPermission(ref, i.keychain, http.DefaultTransport) == nil
+}
+
+func (i *Image) CheckReadAccess() bool {
+	_, err := i.headManifest()
+	if err != nil {
+		if transportErr, ok := err.(*transport.Error); ok {
+			return transportErr.StatusCode != http.StatusUnauthorized &&
+				transportErr.StatusCode != http.StatusForbidden
+		}
+		return false
+	}
+	return true
+}
+
+func (i *Image) headManifest() (*v1.Descriptor, error) {
```

Comment:
```suggestion
func (i *Image) found() (*v1.Descriptor, error) {
```

Love all this! Let's rename this to lowercase `found()`.
Having a similarly named **Uppercase/lowercase** methods is a (completely optional) golang idiom in these types of cases.

Reply:
I addressed this suggestion in the following [commit](https://github.com/buildpacks/imgutil/pull/123/commits/d588cd706db0e32f55858fa00211c948d045210a)